internal/pkg/cli/deploy/deploy.go

```diff
@@ -1197,35 +1229,47 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
 	return nil
 }
 
-func (d *lbSvcDeployer) validateALBWSRuntime() error {
-	if d.lbMft.RoutingRule.Alias.IsEmpty() {
-		if d.env.HasImportedCerts() {
+type validateALBRuntimeInput struct {
+	ServiceName        string
+	RoutingRule        manifest.RoutingRuleConfigOrBool
+	App                *config.Application
+	Env                *config.Environment
+	AppVersionGetter   versionGetter
+	AliasCertValidator aliasCertValidator
+}
+
+func validateALBRuntime(in *validateALBRuntimeInput) error {
+	if in.RoutingRule.Alias.IsEmpty() {
+		if in.Env.HasImportedCerts() {
 			return &errSvcWithNoALBAliasDeployingToEnvWithImportedCerts{
-				name:    d.name,
-				envName: d.env.Name,
+				name:    in.ServiceName,
+				envName: in.Env.Name,
```

Comment:
Blame me but there's seem to be a bug. When http is disabled (or not set for backend svc)

```yaml
http: false
```

We shouldn't error out saying if any cert is imported, `alias` has to be set.